### PR TITLE
nodejs-16.yaml: Add openssf-options

### PR DESCRIPTION
**Pull request by justinvreeland**

### CODE REVIEW

**Review by dannf — COMMENTED** [review]
Questions inline, but no objection to merging as-is.

File: nodejs-16.yaml [hunk 1]
```diff
@@ -45,6 +46,9 @@ pipeline:
       # Add defines recommended in libuv readme.
       common_flags="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
 
+      # Fix the standard to work with newer code
+      sed s/gnu++14/gnu++17/g -i ./common.gypi
```
[the remark this answers is not in the document]
Comment (justinvreeland) [reply nodejs-16.yaml:50]: Ahh that is fair in the future I'll gate things like this with an if ... grep or similar. To be honest I don't totally understand why we suddenly needed a newer standard version here other than the code seems to be using it.

**Review by dannf — APPROVED** [review]
At any rate, nothing noted is a regression, so +1

File: nodejs-16.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mamccorm) [reply nodejs-16.yaml:50]: Believe I am also hitting this for node-22

**Pull request merged**